library/common/config_template.cc

```diff
@@ -51,8 +51,45 @@ R"(
       endpoints: *base_endpoints
     tls_context: *base_tls_context
     type: LOGICAL_DNS
+  - name: stats
```

Comment:
I would like to avoid more involved templating for all of this so I feel fine about the library having this, and the config below, even if people are not interested in flushing metrics. However, open to discussion on if we want to opt in this body of config.